test.js

```diff
@@ -1868,18 +1884,32 @@ testParseError('Invalid header flag bits, must be 0x0 for puback packet', Buffer
   0, 2 // Message ID
 ]))
 
+testParseGenerate('puback without reason and no MQTT 5 properties', {
+  cmd: 'puback',
+  retain: false,
+  qos: 0,
+  dup: false,
+  length: 2,
+  messageId: 2,
+  reasonCode: 0
+}, Buffer.from([
+  64, 2, // Header
+  0, 2 // Message ID
+]), { protocolVersion: 5 })
+
 testParseGenerate('puback with reason and no MQTT 5 properties', {
   cmd: 'puback',
   retain: false,
   qos: 0,
   dup: false,
-  length: 3,
+  length: 4,
```

Comment:
Shouldn't we also keep tha old test with length 3 and no user properties bit?

Reply:
You are right - this test is already in place - however the diff is misleading; take a closer look!
https://github.com/mqttjs/mqtt-packet/pull/142/files#diff-13876b4beb64b9f156474dc78f9c923952a7ca210d4507b6b3135bbe244f8a60R737-R751

Reply:
@robertsLando, btw - thank you for taking the time to guide me through this - as already mentioned I'm absolutely no JS expert.

Reply:
Thanks to you for the PR :) BTW I saw it now, thanks!